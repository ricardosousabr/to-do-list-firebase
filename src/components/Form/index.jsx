import { useState } from 'react'

import { initializeApp } from 'firebase/app'
import { getDatabase, ref, set } from 'firebase/database'
import firebaseConfig from '../../../firebase/config.js'

export default function Form() {
  const [valueInput, setValueInput] = useState('')
  const app = initializeApp(firebaseConfig)
  const database = getDatabase()

  function handleSubmit(e) {
    e.preventDefault()
    sendData()
  }

  function sendData() {
    if (valueInput != '') {
      set(ref(database, 'task/' + 1), {
        task: valueInput
      })
    } else {
      alert('Preencha todos os campos')
    }
  }

  return (
    <div>
      <div>
        <form onSubmit={handleSubmit}>
          <input
            value={valueInput}
            type="text"
            onChange={(e) => setValueInput(e.target.value)}
          />
          <button>click</button>
        </form>
      </div>
    </div>
  )
}
