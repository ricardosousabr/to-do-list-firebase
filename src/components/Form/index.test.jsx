/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import Form from './index'

it('renders homepage unchanged', () => {
  const { container } = render(<Form />)
  expect(container).toMatchSnapshot()
})
