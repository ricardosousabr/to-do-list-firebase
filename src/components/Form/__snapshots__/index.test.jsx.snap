// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div>
    <div>
      <form>
        <input
          type="text"
          value=""
        />
        <button>
          click
        </button>
      </form>
    </div>
  </div>
</div>
`;
